strawberryfields/engine.py

```diff
@@ -556,21 +551,40 @@ def run(self, program: Program, shots: Optional[int] = None) -> Optional[Result]
             self._connection.cancel_job(job.id)
             return None
 
-    def run_async(self, program: Program, shots: Optional[int] = None) -> Job:
+    def run_async(self, program: Program, *, compile_options=None, **run_options) -> Job:
         """Runs a non-blocking remote job.
 
         In the non-blocking mode, a ``Job`` object is returned immediately, and the user can
         manually refresh the status and check for updated results of the job.
 
         Args:
             program (strawberryfields.Program): the quantum circuit
+            compile_options (None, Dict[str, Any]): keyword arguments for :meth:`.Program.compile`
+
+        Keyword Args:
             shots (Optional[int]): The number of shots for which to run the job. If this
                 argument is not provided, the shots are derived from the given ``program``.
 
         Returns:
             strawberryfields.api.Job: the created remote job
         """
-        return self._connection.create_job(self.target, program, shots)
+        # get the specific chip to submit the program to
+        # TODO: this should be provided by the chip API, rather
+        # than built-in to Strawberry Fields.
+        target = self.DEFAULT_TARGETS.get(self.target, self.target)
```

Comment:
Just for clarification here: if a chip family was defined (in `self.target`), then get the default specific chip, else a specific chip was defined (in `self.target`) so we simply need to use that, correct?

Reply:
Yep! It does look a bit weird, but essentially:

* If `self.target` is a DEFAULT_TARGET key, replace it with the corresponding default value.

* Else, just use the value of `self.target`.

Note that validation of `self.target` is already done in `__init__`, hence why it is not repeated here.

Reply:
Maybe I'm missing something, but I cannot find `self.target` as an attribute, only `self._target`. Is this a typo?

Also, shouldn't the validation of `self._target` also allow for the `DEFAULT_TARGET` keys as well (otherwise an error will be raised when initializing with `X8` or `X12`, since they are not in `VALID_TARGETS`).

Reply:
Ignore my first comment. I just missed the `@property` being defined. Sorry.

Reply:
Thanks for catching that @thisac, have fixed it in https://github.com/XanaduAI/strawberryfields/pull/337/files#r399648997 and added tests!